File: src/main/java/io/alchemynft/attestation/RawSignature.java
package io.alchemynft.attestation;

import org.bouncycastle.crypto.AsymmetricCipherKeyPair;

public class RawSignature extends AbstractSignature {
  private static final String TYPE_OF_SIGNATURE = "raw";

  public RawSignature(AsymmetricCipherKeyPair keys, byte[] unprocessedMsg) {
    super(keys, unprocessedMsg, TYPE_OF_SIGNATURE);
  }

  public RawSignature(byte[] signature) {
    super(signature, TYPE_OF_SIGNATURE);
  }

  @Override
  public byte[] processMessage(byte[] unprocessedMsg) {
    return unprocessedMsg.clone();
  }
}

File: src/main/javascript/crypto/src/libs/RawSignature.ts
import {AbstractSignature} from "./AbstractSignature";
import {KeyPair} from "./KeyPair";
import {SignatureUtility} from "./SignatureUtility";
import {uint8toString} from "./utils";

export class RawSignature extends AbstractSignature {
    static TYPE_OF_SIGNATURE = "raw";

    public fromMessage(keys: KeyPair, unprocessedMsg: Uint8Array) {
        super.keyMessageType(keys, unprocessedMsg, RawSignature.TYPE_OF_SIGNATURE);
    }

    public fromSignature( rawSignature: string) {
        super.signatureAndType(rawSignature, RawSignature.TYPE_OF_SIGNATURE);
    }

    public processMessage( unprocessedMsg: Uint8Array): number[] {
        return Array.from(unprocessedMsg);
    }

}